File: ddingmin/baekjoon/7983.py
import sys


def solve(n, arr):
    ans = float('inf')
    for doing, deadline in arr:
        ans = min(ans, deadline) - doing

    return ans


def main():
    n = int(input())
    arr = sorted([list(map(int, input().split())) for _ in range(n)], key=lambda x: [-x[1]])
    print(solve(n, arr))


if __name__ == '__main__':
    input = sys.stdin.readline
    main()

File: ddingmin/baekjoon/7983.kt
import java.io.BufferedWriter
import java.io.OutputStreamWriter

fun main() = with(System.`in`.bufferedReader()) {
    val sout = BufferedWriter(OutputStreamWriter(System.out))

    var n = readLine().toInt()
    var arr = Array(n) {
        readLine().split(" ").map { it.toLong() }
    }

    sout.appendLine(solve(n, arr))
    sout.close()
}

fun solve(n: Int, arr: Array<List<Long>>): String {
    var ans = Long.MAX_VALUE
    arr.sortBy { -it[1] }

    for (values in arr) {
        var doing = values[0]
        var deadline = values[1]
        ans = minOf(ans, deadline) - doing
    }


    return ans.toString()
}
